velox/functions/prestosql/aggregates/tests/AggregationTestBase.h

```diff
@@ -16,13 +16,74 @@
 #pragma once
 
 #include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
 
 namespace facebook::velox::aggregate::test {
 
 class AggregationTestBase : public exec::test::OperatorTestBase {
  protected:
   std::vector<RowVectorPtr>
   makeVectors(const RowTypePtr& rowType, vector_size_t size, int numVectors);
+
+  /// Generates a variety of logically equivalent plans to compute aggregations
+  /// using combinations of partial, final, single, and intermediate
+  /// aggregations with and without local exchanges. Runs all these plans and
+  /// verifies results against DuckDB.
+  ///
+  /// The following plans are generated:
+  ///     - partial -> final
+  ///     - single
+  ///     - partial -> intermediate -> final
+  ///     - partial -> local exchange -> final
+  ///     - partial -> local exchange -> intermediate -> local exchange -> final
+  ///
+  /// Example:
+  ///     testAggregations(
+  ///         [&](PlanBuilder& builder) {
+  ///             builder
+  ///                 .values(vectors)
+  ///                 .project({"c0 % 10 AS c0_mod_10", "c1"});
+  ///         },
+  ///      {"c0_mod_10"},
+  ///      {"sum(c1)"},
+  ///      "SELECT c0 % 10, sum(c1) FROM tmp GROUP BY 1");
+  ///
+  /// @param makeSource A function taking a reference to PlanBuilder and
+  /// building up the plan all the way up to the aggregation.
+  /// @param groupingKeys A list of grouping keys. May be empty.
+  /// @param aggregates A list of aggregates to compute.
+  /// @param duckDbSql An equivalent DuckDB SQL query.
+  void testAggregations(
+      std::function<void(exec::test::PlanBuilder&)> makeSource,
+      const std::vector<std::string>& groupingKeys,
+      const std::vector<std::string>& aggregates,
+      const std::string& duckDbSql);
+
+  /// Same as above, but allows to specify a set of projections to apply after
+  /// the aggregation.
+  void testAggregations(
+      std::function<void(exec::test::PlanBuilder&)> makeSource,
+      const std::vector<std::string>& groupingKeys,
+      const std::vector<std::string>& aggregates,
+      const std::vector<std::string>& postAggregationProjections,
```

Comment:
Should we combine the two APIs and provide a default for the last argument `const std::vector<std::string>& postAggregationProjections = {}` ?

Same for the two below.

Reply:
I felt that usage would be awkward, e.g.

```
    testAggregations(
        rowVectors,
        {"c0"},
        {"checksum(c1)"},
        {"to_base64(a0)"},
        expectedDuckDbSql);
```

vs. 

```
    testAggregations(
        rowVectors,
        {"c0"},
        {"checksum(c1)"},
        expectedDuckDbSql,
        {"to_base64(a0)"});
```

Reply:
Makes sense. This is fine then.